examples/samples/timeToLive.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  delay,
+  OnMessage,
+  OnError,
+  MessagingError,
+  ServiceBusMessage,
+  ReceiveMode,
+  Namespace,
+  SendableMessageInfo,
+  generateUuid
+} from "../../lib";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const queuePath = process.env.QUEUE_NAME || "";
+const deadLetterQueuePath = queuePath + "/$DeadLetterQueue";
+const receiveClientTimeoutInMilliseconds = 10000;
+console.log("str: ", str);
+console.log("queue path: ", queuePath);
+console.log("deadletter queue path: ", deadLetterQueuePath);
+
+let ns: Namespace;
+
+/*
+  This sample demonstrates how TimeToLive property works. For more information, refer to
+  https://docs.microsoft.com/en-us/azure/service-bus-messaging/message-expiration
+
+  On running this sample, you should immediately see a Pineapple Icecream message in the queue.
+  After 30 seconds, you should see the message moved to DLQ with appropriate reason being set.
+*/
+async function main(): Promise<void> {
+  try {
+    ns = Namespace.createFromConnectionString(str);
+    // Send messages with TTL property assigned
+    await SendMessagesAsync();
+
+    // Expire the messages
+    await delay(60000);
+
+    // Process expired messages from the Dead Letter Queue
+    await PickUpAndFixDeadletters();
+  } catch (err) {
+    console.log(">>>>> Error occurred in running sampple: ", err);
+  } finally {
+    console.log("\n>>>> Calling close....");
+    ns.close();
+  }
+}
+
+// Task to send messages to given queue
+async function SendMessagesAsync(): Promise<void> {
+  const recipeData = { name: "Limited Pineapple Icecream", type: "Dessert" };
+  const client = ns.createQueueClient(queuePath);
+  const messageBody = recipeData;
+  const message: SendableMessageInfo = {
+    body: messageBody,
+    contentType: "application/json",
+    label: "Recipe",
+    timeToLive: 30 * 1000, // 30 seconds
+    messageId: generateUuid()
+  };
+  await client.send(message);
+  await client.close();
+}
+
+// Task for processing the expired messages after they land in the Dead Letter Queue
+async function PickUpAndFixDeadletters(): Promise<void> {
```

Comment:
As a user, I would expect the demonstration of time to live feature would be to
- let the message expire
- prove that normal way of receiving messages doesnt receive the messages, well because they expired
- prove that the expired messages are in the dead letter queue 

The whole concept of fixing these messages and re-submitting feels un-necessary for this sample. It may have a place in the sample for DeadLetter queue, but not for Time To Live 

Thoughts?

cc @ShivangiReja , @ramya0820 , @AlexGhiondea, @HarshaNalluru

Reply:
Yes, I think for this sample fixing and resubmitting is not necessary because we trying to make these samples as simple as possible for the users, so there is no need to add extra functionality.